sync/block_listener.go

```diff
@@ -132,12 +133,13 @@ func (bl *BlockListener) run() {
 //todo handle case where no peer knows the block
 func (bl *BlockListener) FetchBlock(id types.BlockID) {
 	for _, p := range bl.GetPeers() {
-		if ch, err := sendBlockRequest(bl.MessageServer, p, id, bl.Log); err == nil {
+		ch, foo := blockRequest()
+		if err := bl.SendRequest(BLOCK, id.ToBytes(), p, foo); err == nil {
 			block := <-ch
 			if block == nil {
 				continue
 			}
-			eligible, err := bl.BlockEligible(block)
+			eligible, err := bl.BlockEligible(&block.BlockHeader)
```

Comment:
Does this check whether the synced block was eligible in its time? why are we panicking if it wasn't ?

Reply:
@noamnelke

Reply:
yes, this is what it checks. It should return an error (that's why there's `TODO: return error` there) but the surrounding method is called from within a go routine, so I left it as technical debt.